server.go

```diff
@@ -511,7 +516,8 @@ func (s *Server) Serve(lis net.Listener) error {
 				timer := time.NewTimer(tempDelay)
 				select {
 				case <-timer.C:
-				case <-s.ctx.Done():
+				case <-s.quit:
+					return nil
 				}
 				timer.Stop()
```

Comment:
Not in your PR but we can get rid of this time.Stop() now that we're here.

Reply:
Moved to the appropriate place.